added new wireguard adapter to latest

Grizzelbee opened the pull request.

Grizzelbee committed:
added new wireguard adapter to latest

Apollon77 commented:
RE-CHECK!

GermanBluefox commented:
## Automated adapter checker

### [ioBroker.wireguard](https://github.com/Grizzelbee/ioBroker.wireguard)
[![Downloads](https://img.shields.io/npm/dm/iobroker.wireguard.svg)](https://www.npmjs.com/package/iobroker.wireguard) ![Number of Installations (latest)](http://iobroker.live/badges/wireguard-installed.svg) 
[![NPM](https://nodei.co/npm/iobroker.wireguard.png?downloads=true)](https://nodei.co/npm/iobroker.wireguard/)

:thumbsup: No errors found
- [ ] :eyes: [W400] Cannot find "wireguard" in latest repository


*Add comment "RE-CHECK!" to start check anew*

Apollon77 commented:
Hi, some review comments:
* if only linux is supported please also add thet to package.json as limitation
* Please move the admin dep to "globalDependencies" https://github.com/Grizzelbee/ioBroker.wireguard/blob/main/io-package.json#L139

Rest seems fine :-) Please check and fix

Grizzelbee commented:
@Apollon77 
> if only linux is supported please also add thet to package.json as limitation
Added windows support as a side effect by dropping usage of wg-json script in latest version. :)

But for curiosity: The windows/linux support applied not to the adapter itself but to the queried wireguard server. Is it possible to model this in the package.json too?
> Please move the admin dep to "globalDependencies"

done.


Changelog:
### v0.9.5 (2022-02-22)
* (grizzelbee) New: dropped use of wg-json script - not needed anymore
* (grizzelbee) New: making internal use of wg show all dump command and self parsing the result
* (grizzelbee) New: Added windows support by using the wg show all command
* (grizzelbee) Upd: moved dependency **admin** to globalDependency as requested during adapter review

Apollon77 commented:
> Btw: What category should I choose/ will be better? Infrastructure or network?

my feeling would glo to "infrastructure" ... but check the other adapters in both groups and maybe decided then

Grizzelbee commented:
In that case I'll stick with "infrastucture".  Then I'm done with all fixes.

Apollon77 commented:
welcome in the repo
